src/target/riscv/riscv-013.c

```diff
@@ -4702,7 +4702,8 @@ static int riscv013_get_register(struct target *target,
 	if (rid == GDB_REGNO_PC) {
 		/* TODO: move this into riscv.c. */
 		result = register_read_direct(target, value, GDB_REGNO_DPC);
-		LOG_TARGET_DEBUG(target, "read PC from DPC: 0x%" PRIx64, *value);
+		if (result == ERROR_OK)
+			LOG_TARGET_DEBUG(target, "read PC from DPC: 0x%" PRIx64, *value);
```

Comment:
If the read succeeds, the register name and value is already printed inside `register_read_direct()` for any register (riscv013.c on line 1722).

So you may consider removing this extra duplicate print of PC value altogether. I would actually prefer that -- for the sake of consistency.

However, if decide to keep the print and have the PC value in the log twice (for extra visibility), then the change looks all right to me.

Reply:
Removed. Thanks.